Source/EasyNetQ/Producer/PersistentChannel.cs

```diff
@@ -31,104 +30,58 @@ IEventBus eventBus
             Preconditions.CheckNotNull(configuration, "configuration");
             Preconditions.CheckNotNull(eventBus, "eventBus");
 
-            this.connection = connection;
             this.configuration = configuration;
             this.eventBus = eventBus;
 
-            WireUpEvents();
+            //Do not cache exceptions and do not allow to call CreateModel concurrently
+            lazyChannel = new Lazy<IModel>(() =>
```

Comment:
`Lazy` and `lock` code smell, lazy should already guarantee the thread safety, but I don't know the purpose of `LazyThreadSafetyMode.PublicationOnly`

Reply:
why not simply use `LazyThreadSafetyMode.ExecutionAndPublication` that actually adds the lock you added inside with a double-check implementation?

Reply:
Because it caches an exception: if the factory will throw an exception during initialisation, the next accesses to Lazy will throw that exception.

Reply:
but in this case in concurrency you could create two channel isn't it?

> Any instances of T that were created by the competing threads are discarded.

so they are discarded, but not disposed so they remain as channel on rmq server, right?

Reply:
>so they are discarded, but not disposed so they remain as channel on rmq server, right?

yeap, you are right. will fix it. thanks 👍

Reply:
Actually, there is no way to create channel here multiple threads, because `PersistentChannel` is used only in single threaded mode: you can see it in `ClientCommandDispatcherSingleton`.